### ISSUE TITLE
Външна analytics услуга vs. in-house решение

### ISSUE THREAD

**Issue by mitio** [governmentbg/opendata#20]
Това е issue, в което да се обсъди дали този сайт, както и потенциално други държавни сайтове, да разчитат на външни услуги за брояч и анализ на трафика (например [Google Analytics](http://www.google.com/analytics/), или да се ползва in-house решение, което извлича и анализира лог-файловете от уеб сървъра, като опционално се сложи и някакъв JavaScript, който да дава допълнителната информация, която инструменти като Google Analytics предоставят.

Срок: **безсрочно**
Приоритет: **нисък**

Тази дискусия започна в #18 и част от аргументите и притесненията да се ползва Google Analytics са изказани там.

В момента opendata.government.bg използва Google Analytics като бързо, лесно, безплатно, добро и цялостно решение, признато от индустрията, но това означава, че освен българското правителство, и Google получават информация за това кой е посещавал сайта и какво е достъпвал там. Това би могло да се избегне с in-house решение.
## Piwik

Беше предложено да се използва [piwik](http://piwik.org/), което изглежда съвсем прилично.

Информация и плюсове:
- [Piwik Demo](http://demo.piwik.org/).
- [GitHub repo](https://github.com/piwik/piwik).
- Има превод на български.
- Поддържа повечето [съществени функционалности](http://piwik.org/features/), вкл. и log analytics. Има и опция за инсталация на JS брояч, както и Google Analytics.
- Technical requirements - PHP 5.3.3 or greater, MySQL 4.1 or greater, and either MySQLi or PDO library must be enabled.
- Опция за self-hosted инсталация, с отворен код, лиценз GPL v3.

Минуси:
- Изисква собствена, отделна инсталация, но това не е голям проблем.
- Не е ясно ако има много трафик, колко performant ще е (на PHP е).
- Не е ясно ако базата данни нарастне, как Piwik ще се справя с това (базата е MySQL).
- Като функционалности и потребителски интерфейс отстъпва на Google Analytics.

/cc @Glamdring @ignisf @krokodilerian и който още има интерес.

**Comment by antitoxic** [comment]
Някой дали потърси "ckan Google analytics"?

Първи резултат:

https://github.com/ckan/ckanext-googleanalytics

:)

**Comment by mitio** [comment]
@antitoxic Това е интересно решение основно заради обратната информация, която вмъква данни от Google Analytics в страници в портала. Още един плюс за Google Analytics.

**Comment by krokodilerian** [comment]
Практически наблюдения от друго място:
- не можем да бием по функционалност google analytics с някой piwik или нещо такова, по никакъв начин.
- всичко такова е още нещо, което бави зареждането (в повечето случаи, дори да се накове асинхронно). Не е ясно дали можем да бием по скорост analytics, и дали можем да докараме същото redundancy.
- простия анализ на логове няма достатъчно информация (колкото може да се вади с javascript и т.н.).

Като цяло, аз най-много мразя всякакви такива външни услуги. Това да обясняваме на google кой би browse-ва е грубо, грозно и политически проблем. От друга страна ако наистина, ама НАИСТИНА имаме нужда от това и не е проблем за потребителите, от това по-свестен вариант няма.

**Comment by jorr** [comment]
Честно казано, знам че сме за shiny графиките и т.н.
Но на нас не ни трябва и 1/100 от функционалността на Г.А.
Трябва ни единствено и само кой дейтасет колко се ползва, поне на първо
време
И то това даже минава повече през АПИ-то, отколкото през page loads.

Определено inhouse решение на бекенда = една табличка в базата и пет реда в
някой контролер е най-смисленото нещо

2015-07-15 17:12 GMT+03:00 Vasil Kolev notifications@github.com:

**Comment by Glamdring** [comment]
@antitoxic да, видях го това, ама добавянето на 1 скрипт в темплейта беше доста по-лесно от слагането на добавка :)

**Comment by ignisf** [comment]
А това ![0csyg9g](https://cloud.githubusercontent.com/assets/1153188/8889282/715555fc-32dc-11e5-8862-15afd2a042be.png)

как се връзва с
http://lex.bg/en/laws/ldoc/2135530547#i_231

**Comment by antitoxic** [comment]
@ignisf #21 Pull request се приемат.

**Comment by ignisf** [comment]
Питам, защото ако няма правно основание да се уведомяват потребителите, че сайтът използва бисквитки, няма нужда да се въведе това парче бюрокрация в UX-а, а аз нямам нужните познания, за да се изкажа.

**Comment by antitoxic** [comment]
@ignisf и аз не съм наясно. Не съм виждал (или ако съм виждал много малко) сайтове в бг да предупреждавата за cookies. Знам че има задължение в EU за известия, когато има бисквитки, но не знам как е интерпретирано и внедрено в БГ.
